[Curl][Soup] Remove CredentialStorage[Curl|Soup].cpp

kshukuwa opened the pull request:
#### fea01c276de6bf0567541cb50802650ad88ac486
<pre>
[Curl][Soup] Remove CredentialStorage[Curl|Soup].cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=259135">https://bugs.webkit.org/show_bug.cgi?id=259135</a>

Reviewed by Fujii Hironori.

CredentialStorage::getFromPersistentStorage is not used in WebKit2.
So curl and soup port don&apos;t need the function. Therefore, to reduce
maintenance costs, remove CredentialStorage[Curl|Soup].cpp.

* Source/WebCore/platform/Curl.cmake:
* Source/WebCore/platform/SourcesSoup.txt:
* Source/WebCore/platform/network/CredentialStorage.h:
* Source/WebCore/platform/network/curl/CredentialStorageCurl.cpp: Removed.
* Source/WebCore/platform/network/soup/CredentialStorageSoup.cpp: Removed.

Canonical link: <a href="https://commits.webkit.org/265988@main">https://commits.webkit.org/265988@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bbdf799ae8aa0a19b31fdd95ef00f1c65682db54

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/12493 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/12823 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/13137 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/14235 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/11989 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/12553 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/15256 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/12842 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/14678 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/12657 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/13397 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/10562 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/14660 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/10676 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/11305 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/18401 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/11764 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/11469 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/14663 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/11956 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/9880 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/11195 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3069 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/15526 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/11811 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

fujii commented on Source/WebCore/platform/network/CredentialStorage.cpp:
```diff
@@ -187,4 +187,11 @@ void CredentialStorage::clearCredentials()
+    return Credential();
+}
+#endif
+
```
I think we should change CredentialStorage.h.

~~~
#if PLATFORM(COCOA)
    // OS credential storage.
    WEBCORE_EXPORT static Credential getFromPersistentStorage(const ProtectionSpace&);
#endif
~~~

kshukuwa committed:
[Curl][Soup] Remove CredentialStorage[Curl|Soup].cpp
https://bugs.webkit.org/show_bug.cgi?id=259135

Reviewed by Fujii Hironori.

CredentialStorage::getFromPersistentStorage is not used in WebKit2.
So curl and soup port don't need the function. Therefore, to reduce
maintenance costs, remove CredentialStorage[Curl|Soup].cpp.

* Source/WebCore/platform/Curl.cmake:
* Source/WebCore/platform/SourcesSoup.txt:
* Source/WebCore/platform/network/CredentialStorage.h:
* Source/WebCore/platform/network/curl/CredentialStorageCurl.cpp: Removed.
* Source/WebCore/platform/network/soup/CredentialStorageSoup.cpp: Removed.

Canonical link: https://commits.webkit.org/265988@main
